src/main/java/edu/usf/cutr/tba/utils/StringUtils.java

```diff
@@ -84,4 +85,50 @@ public static String validateAndParseOutputPath(String newPath) {
         }
     }
 
+    /**
+     * If stringTimestamp is parsable to Long then return its long value, it it is not parsable, the function will find
+     * the best timeStamp option between a prefix value of the provided string or the MIN_VALUE of the Long class.
+     * @param timeStamp string containing a possible time stamp
+     * @return Long time stamp
+     */
+    public static long timeStampToLong(String timeStamp) {
+        if (!NumberUtils.isParsable(timeStamp)) {
+            // The timestamp must be a random string with a numeric prefix, get the prefix
+            int indexDash = timeStamp.indexOf("-");
+            if (indexDash == -1) {
+                // Dash not found, return MIN_VALUE
+                return Long.MIN_VALUE;
+            }
+            timeStamp = timeStamp.substring(0 , indexDash);
```

Comment:
The prefix isn't a timestamp, so I'd avoid assigning it to the `timestamp` variable here to avoid future confusion.

Reply:
I added a local variable for the prefix. substring.